openstef/pipeline/optimize_hyperparameters.py

```diff
@@ -62,6 +63,64 @@ def optimize_hyperparameters_pipeline(
     Returns:
         dict: Optimized hyperparameters.
     """
+
+    (
+        best_model,
+        model_specs,
+        report,
+        trials,
+        best_trial_number,
+        best_params,
+    ) = optimize_hyperparameters_pipeline_core(pj, input_data, horizons, n_trials)
+
+    # Create serializer
+    serializer = MLflowSerializer(mlflow_tracking_uri=mlflow_tracking_uri)
+
+    # Save model, optimization results and report
+    serializer.save_model(
+        model=best_model,
+        experiment_name=str(pj["id"]),
+        model_type=pj["model"],
+        model_specs=model_specs,
+        report=report,
+        phase="Hyperparameter_opt",
+        trials=trials,
+        trial_number=best_trial_number,
+    )
+    if artifact_folder:
+        Reporter.write_report_to_disk(report=report, artifact_folder=artifact_folder)
+    return best_params
+
+
+def optimize_hyperparameters_pipeline_core(
```

Comment:
Nice to split the pipeline into a core without model saving and a pipeline with model saving! Can you add one simple unit test for optimize_hyperparameters_pipeline_core? Just one simple unit test will do in my opinion and probs existing tests can be reused :)